gossip/src/cluster_info.rs

```diff
@@ -1035,14 +1056,18 @@ impl ClusterInfo {
                 .min() // Boot the oldest evicted vote by wallclock.
                 .map(|(_ /*wallclock*/, ix)| ix)
         };
+        if is_duplicate_vote {
+            // Don't do anything. Perhaps in the future we can use this indicator to adopt a newer
+            // vote state from gossip.
+            return;
+        }
         let vote_index = vote_index.unwrap_or(num_crds_votes);
         if (vote_index as usize) >= MAX_LOCKOUT_HISTORY {
```

Comment:
Not sure any of the code changes above this line are necessary.
It will suffice just to change `panic!` here to an error log or something, and then return.

Reply:
The issue is that this panic doesn't actually occur during the duplicate vote, but later once `gossip_crds` gets filled.

Reply:
If we don't care about differentiating failure conditions we could remove all this code and change the panic to info, however we would also lose the log message for the duplicate vote.

Reply:
@AshwinSekar is the example you're illustrating here something like if we push a vote for slot `X`, and then on restart, we push another version of a vote for slot `X`, such that now slot `X` takes two spots in gossip. Then when we fill to the remaining 30 votes in the tower, we'll panic, because a slot in the tower has taken two slots in gossip

Reply:
Exactly. It's not the duplicate vote that'll cause the panic, but down the line because we have too many votes.

Reply:
Probably worth logging for this case b/c it's a bit complex